src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -535,5 +493,128 @@ private void RemoveValueFromFile<T>(string fileName, string key)
+    ///      SubKeys                  Name-Value-Pairs
+    ///      - ScriptBlockLogging     { EnableScriptBlockLogging (0 or 1); EnableScriptBlockInvocationLogging (0 or 1) }
+    ///      - ModuleLogging          { EnableModuleLogging (0 or 1); ModuleNames (string[]) }
+    ///      - Transcription          { EnableTranscripting (0 or 1); OutputDirectory (string); EnableInvocationHeader (0 or 1) }
+    ///      - UpdatableHelp          { DefaultSourcePath (string) }
+    ///      - ConsoleSessionConfiguration { EnableConsoleSessionConfiguration (0 or 1); ConsoleSessionConfigurationName (string) }
+    ///  - Software\Policies\Microsoft\Windows\EventLog
+    ///     SubKeys                   Name-Value-Pairs
+    ///      - ProtectedEventLogging  { EnableProtectedEventLogging (0 or 1); EncryptionCertificate (string[]) }
+    ///
+    /// The JSON representation is in sync with the 'PowerShellPolicies' type. Here is an example:
+    /// {
+    ///   "PowerShellPolicies": {
+    ///     "ScriptExecution": {
+    ///       "ExecutionPolicy": "RemoteSigned"
+    ///     },
+    ///     "ScriptBlockLogging": {
+    ///       "EnableScriptBlockInvocationLogging": true,
+    ///       "EnableScriptBlockLogging": false
+    ///     },
+    ///     "ProtectedEventLogging": {
+    ///       "EnableProtectedEventLogging": false,
+    ///       "EncryptionCertificate": [
+    ///         "Joe"
+    ///       ]
+    ///     },
+    ///     "Transcription": {
+    ///       "EnableTranscripting": true,
+    ///       "EnableInvocationHeader": true,
+    ///       "OutputDirectory": "c:\tmp"
+    ///     },
+    ///     "UpdatableHelp": {
+    ///       "DefaultSourcePath": "f:\temp"
+    ///     },
+    ///     "ConsoleSessionConfiguration": {
+    ///       "EnableConsoleSessionConfiguration": true,
+    ///       "ConsoleSessionConfigurationName": "name"
+    ///     }
+    ///   }
+    /// }
+    /// </summary>
+    internal sealed class PowerShellPolicies
+    {
+        public ScriptExecution ScriptExecution { get; set; }
+        public ScriptBlockLogging ScriptBlockLogging { get; set; }
+        public ModuleLogging ModuleLogging { get; set; }
+        public ProtectedEventLogging ProtectedEventLogging { get; set; }
+        public Transcription Transcription { get; set; }
+        public UpdatableHelp UpdatableHelp { get; set; }
+        public ConsoleSessionConfiguration ConsoleSessionConfiguration { get; set; }
+    }
+
+    internal abstract class PolicyBase { }
+
+    /// <summary>
+    /// Setting about ScriptExecution
+    /// </summary>
+    internal sealed class ScriptExecution : PolicyBase
+    {
+        public string ExecutionPolicy { get; set; }
+        public bool? EnableScripts { get; set; }
+    }
+
+    /// <summary>
+    /// Setting about ScriptBlockLogging
+    /// </summary>
+    internal sealed class ScriptBlockLogging : PolicyBase
```

Comment:
The same - maybe `ScriptBlockLoggingPolicy`?

Reply:
I see check null for instance the classes - maybe static is better if we always process config?

Reply:
We are using the same name as the keys in GPO, which I think will make it easy to correlate them, and also make it easy to read GPO settings on Windows. You will see that in my new commits.

Reply:
> I see check null for instance the classes - maybe static is better if we always process config?

I'm not sure what this comment is referring to. Can you be more specific?

Reply:
Regarding your comment about static; this class should be static. There's nothing in this class that requires an instance and the fields are all read-only in usage.

Reply:
@dantraMSFT Agreed. I left a similar comment at https://github.com/PowerShell/PowerShell/pull/5791#discussion_r159966649. We need to revisit the configuration APIs when working on the experimental feature work.